gvwilson/new-lesson-example
# Sample unit test for lesson validator

## Purpose
Demonstrate unit testing of a function in the validator, to get the ball rolling.

This code is slightly unfamiliar and it's late at night, so please be on the lookout for regressions / bugs. I'll be happy to fix anything noted.

## Summary of changes
- Add a new `make unittest` command to run tests
- Rename the validator python files to something that works more naturally with python imports, and update makefile commands so the difference should not be noticeable to users
- Change call signature of `check_fileset` function to better support unit testing
- Add a single test that verifies no errors when the validator sees the expected list of files
